IBPSA/Resources/Scripts/Dymola/ConvertIBPSA_from_4.0.0_to_4.0.1.mos

```diff
@@ -0,0 +1,790 @@
+    "thermalZoneTwoElements.alphaExt",
+    "thermalZoneTwoElements.hConExt");
+convertElement(
+    "IBPSA.ThermalZones.ReducedOrder.Validation.VDI6007.TestCase1",
+    "thermalZoneTwoElements.alphaExt",
+    "thermalZoneTwoElements.hConExt");
+convertElement(
+    "IBPSA.ThermalZones.ReducedOrder.Validation.VDI6007.TestCase10",
+    "thermalZoneTwoElements.alphaExt",
+    "thermalZoneTwoElements.hConExt");
+convertElement(
+    "IBPSA.ThermalZones.ReducedOrder.Validation.VDI6007.TestCase11",
+    "thermalZoneTwoElements.alphaExt",
+    "thermalZoneTwoElements.hConExt");
+convertElement(
+    "IBPSA.ThermalZones.ReducedOrder.Validation.VDI6007.TestCase12",
+    "thermalZoneTwoElements.alphaExt",
+    "thermalZoneTwoElements.hConExt");
+convertElement(
+    "IBPSA.ThermalZones.ReducedOrder.Validation.VDI6007.TestCase2",
+    "thermalZoneTwoElements.alphaExt",
+    "thermalZoneTwoElements.hConExt");
+convertElement(
+    "IBPSA.ThermalZones.ReducedOrder.Validation.VDI6007.TestCase3",
+    "thermalZoneTwoElements.alphaExt",
+    "thermalZoneTwoElements.hConExt");
+convertElement(
+    "IBPSA.ThermalZones.ReducedOrder.Validation.VDI6007.TestCase4",
+    "thermalZoneTwoElements.alphaExt",
+    "thermalZoneTwoElements.hConExt");
+convertElement(
+    "IBPSA.ThermalZones.ReducedOrder.Validation.VDI6007.TestCase5",
+    "thermalZoneTwoElements.alphaExt",
+    "thermalZoneTwoElements.hConExt");
+convertElement(
+    "IBPSA.ThermalZones.ReducedOrder.Validation.VDI6007.TestCase6",
+    "thermalZoneTwoElements.alphaExt",
+    "thermalZoneTwoElements.hConExt");
+convertElement(
+    "IBPSA.ThermalZones.ReducedOrder.Validation.VDI6007.TestCase7",
+    "thermalZoneTwoElements.alphaExt",
+    "thermalZoneTwoElements.hConExt");
+convertElement(
+    "IBPSA.ThermalZones.ReducedOrder.Validation.VDI6007.TestCase8",
+    "thermalZoneTwoElements.alphaExt",
+    "thermalZoneTwoElements.hConExt");
+convertElement(
+    "IBPSA.ThermalZones.ReducedOrder.Validation.VDI6007.TestCase9",
+    "thermalZoneTwoElements.alphaExt",
+    "thermalZoneTwoElements.hConExt");
+convertElement(
+    "IBPSA.ThermalZones.ReducedOrder.RC.FourElements",
+    "alphaExtWallConst",
+    "hConExtWall_const");
+convertElement(
+    "IBPSA.ThermalZones.ReducedOrder.RC.OneElement",
+    "alphaExtWallConst",
+    "hConExtWall_const");
+convertElement(
+    "IBPSA.ThermalZones.ReducedOrder.RC.ThreeElements",
+    "alphaExtWallConst",
+    "hConExtWall_const");
+convertElement(
+    "IBPSA.ThermalZones.ReducedOrder.RC.TwoElements",
+    "alphaExtWallConst",
+    "hConExtWall_const");
+convertElement(
+    "IBPSA.ThermalZones.ReducedOrder.Examples.SimpleRoomFourElements",
+    "thermalZoneFourElements.alphaExtWallConst",
+    "thermalZoneFourElements.hConExtWall_const");
+convertElement(
+    "IBPSA.ThermalZones.ReducedOrder.Examples.SimpleRoomOneElement",
+    "thermalZoneOneElement.alphaExtWallConst",
+    "thermalZoneOneElement.hConExtWall_const");
+convertElement(
```

Comment:
It looks like you misunderstand conversion scripts: Entries such as
```
convertElement(
    "IBPSA.ThermalZones.ReducedOrder.Examples.SimpleRoomOneElement",
    "thermalZoneOneElement.alphaExtWallConst",
    "thermalZoneOneElement.hConExtWall_const");
```
are not correct (or at least not needed). Because the _instance_ `thermalZoneOneElement` is an instance of the class `IBPSA.ThermalZones.ReducedOrder.RC.OneElement`, the correct entry should be
```
convertElement(
    "IBPSA.ThermalZones.ReducedOrder.RC.OneElement",
    "alphaExtWallConst",
    "hConExtWall_const");
```
This is sufficient, and will cause Dymola to also update the parameter assignment in `IBPSA.ThermalZones.ReducedOrder.Examples.SimpleRoomOneElement`. See also the Dymola manual.

Once you revised the conversion script, please convert an old example to the new library to make sure the script is correct, as it is easy to make mistakes in this file.

Reply:
We have assumed that we should use dymola's automatic generation of conversion scripts. So we prevent overlooking parameters. Originally I wrote a shorter conversion script according to the Dymola manual. After Dymola wrote commands like
```
convertElement(
    "IBPSA.ThermalZones.ReducedOrder.Examples.SimpleRoomOneElement",
    "thermalZoneOneElement.alphaExtWallConst",
    "thermalZoneOneElement.hConExtWall_const");
```
I thought they might be necessary after all. But I get your point and deleted all entries I consider superfluous. Thinking about this: Would you consider the entries about the example and validation models, e.g.
```
convertElement(
    "IBPSA.ThermalZones.ReducedOrder.Examples.SimpleRoomFourElements",
    "alphaWall",
    "hConWall");
```
necessary? I don't think anyone's gonna use these models in another library. But I'm not sure if I'll forget use cases where it would be necessary to execute these commands.
I successfully tested the conversion on a few examples.

Reply:
I don't think it is important to convert parameters in `Examples` or `Validation`, but if you have already done the scripts, we can include them.